git-branchless/tests/test_move.rs

```diff
@@ -4866,3 +4866,1364 @@ fn test_move_delete_branch_config_entry() -> eyre::Result<()> {
+            "move",
+            "--in-memory",
+            "--fixup",
+            "-x",
+            &test2_oid.to_string(),
+        ])?;
+        insta::assert_snapshot!(stdout, @r###"
+        Attempting rebase in-memory...
+        [1/1] Committed as: ff6183f update 3 test.txt
+        branchless: processing 2 rewritten commits
+        branchless: running command: <git-executable> checkout ff6183fc6b71c3fcf6f26247162bde4e34bc5193
+        O f777ecc (master) create initial.txt
+        |
+        o 307a04c create test.txt
+        |
+        @ ff6183f update 3 test.txt
+        In-memory rebase succeeded.
+        "###);
+
+        let diff = git.get_trimmed_diff("test.txt", "HEAD")?;
+        insta::assert_snapshot!(diff, @r###"
+        @@ -1 +1,3 @@
+        +line 1
+         line 2
+        +line 3
+        "###);
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_move_fixup_head_into_ancestor() -> eyre::Result<()> {
+    let git = make_git()?;
+
+    if !git.supports_committer_date_is_author_date()? {
+        return Ok(());
+    }
+    git.init_repo()?;
+
+    git.detach_head()?;
+    let test1_oid = git.commit_file_with_contents("test", 1, "line 2\n")?;
+    let _test2_oid =
+        git.commit_file_with_contents_and_message("test", 2, "line 1\nline 2\n", "update 2")?;
+    let _test3_oid = git.commit_file_with_contents_and_message(
+        "test",
+        3,
+        "line 1\nline 2\nline 3\n",
+        "update 3",
+    )?;
+
+    let (stdout, _stderr) = git.run(&["smartlog"])?;
+    insta::assert_snapshot!(stdout, @r###"
+    O f777ecc (master) create initial.txt
+    |
+    o 307a04c create test.txt
+    |
+    o 6d92723 update 2 test.txt
+    |
+    @ da38dc8 update 3 test.txt
+    "###);
+
+    let diff = git.get_trimmed_diff("test.txt", "HEAD")?;
+    insta::assert_snapshot!(diff, @r###"
+    @@ -1,2 +1,3 @@
+     line 1
+     line 2
+    +line 3
+    "###);
+
+    // --in-memory
+    {
+        let (stdout, _stderr) = git.run(&[
+            "move",
+            "--in-memory",
+            "--fixup",
+            "-s",
+            "HEAD",
+            "-d",
+            &test1_oid.to_string(),
+        ])?;
+        insta::assert_snapshot!(stdout, @r###"
+        Attempting rebase in-memory...
+        [1/2] Committed as: 963fb93 create test.txt
+        [2/2] Committed as: b9da1e0 update 2 test.txt
+        branchless: processing 3 rewritten commits
+        branchless: running command: <git-executable> checkout 963fb93e7a4b7c53a1feb565106a1a4e5a298bf4
+        O f777ecc (master) create initial.txt
+        |
+        @ 963fb93 create test.txt
+        |
+        o b9da1e0 update 2 test.txt
+        In-memory rebase succeeded.
+        "###);
+
+        let diff = git.get_trimmed_diff("test.txt", "HEAD")?;
+        insta::assert_snapshot!(diff, @r###"
+        @@ -0,0 +1,2 @@
+        +line 2
+        +line 3
+        "###);
+        let diff = git.get_trimmed_diff("test.txt", "b9da1e0")?;
```

Comment:
I made this change and was pleased w/ the results ... except that it turned up a bug in the on-disk fixups! (That's a good thing, I think. 🤔) I was able to fix that specific test failure, but it feels a bit specific and hacky, and I don't have capacity right now to investigate it in depth for a proper fix.

Based on that, I've backed out the last commit in this stack. That leaves this PR with in memory fixups intact, but on disk fixups are `unimplemented!()`. I'll push my on-disk work to a separate PR.

Reply:
Resolving this, as all of the new tests in this PR use branches instead of commit oids. If you'd like me to do something else instead of `unimplemented!()` for the on-disk fixups, please let me know.